File: backend/internal/constants/stock_alert.go
package constants

// AlertView 预警通知视图：未读 / 全部，前后端共享定义
// （frontend/src/constants/stockAlert.ts 对应实现）。
type AlertView string

const (
	AlertViewUnread AlertView = "unread"
	AlertViewAll    AlertView = "all"
)

// NormalizeAlertView 将入参归一为合法视图，默认未读。
func NormalizeAlertView(v AlertView) AlertView {
	switch v {
	case AlertViewUnread, AlertViewAll:
		return v
	}
	return AlertViewUnread
}

File: frontend/src/constants/stockAlert.ts
// 库存预警通知视图枚举（与 backend/internal/constants/stock_alert.go 对应）
export const AlertView = {
  UNREAD: 'unread',
  ALL: 'all'
} as const

export type AlertViewValue = (typeof AlertView)[keyof typeof AlertView]

export const ALERT_VIEW_OPTIONS: { label: string; value: AlertViewValue }[] = [
  { label: '未读', value: AlertView.UNREAD },
  { label: '全部', value: AlertView.ALL }
]
